### PR TITLE
KK-123 | Adjustments for small screens and modal fixes

### PR DESCRIPTION
**Pull request by badrange**
* Move add child button below child list
* Profile page edit button aligns as well as possible with name
* Hide back button on child detail page
* Modal has correct button order on small screen and correct size on big screen

### CODE REVIEW

File: src/common/components/modal/modal.module.scss [hunk 1]
```diff
@@ -18,6 +19,9 @@ $modalPadding: 2rem;
     right: 0;
     bottom: 0;
   }
+  @include respond-above(md) {
```
Comment (midori0507) [inline src/common/components/modal/modal.module.scss]: i think this is safer
https://getbootstrap.com/docs/4.2/components/modal/#optional-sizes

Comment (midori0507) [reply src/common/components/modal/modal.module.scss]: maybe proportional to breakpoint values

File: src/common/components/modal/modal.module.scss [hunk 2]
```diff
@@ -32,35 +36,42 @@ $modalPadding: 2rem;
...
+    position: relative;
+    text-align: right;
+    width: 100%;
+  }
+  button {
+    padding: 0;
+    border: none;
+    height: $largeMargin;
+    width: $largeMargin;
+    font-weight: $font-weight-bold;
```
Comment (midori0507) [inline src/common/components/modal/modal.module.scss]: The x button is not a character, its an image, so this is useless (try to toggle it on and off)

You should control the image size instead if you expect the same effect

File: src/common/components/modal/modal.module.scss [hunk 3]
```diff
@@ -32,35 +36,42 @@ $modalPadding: 2rem;
...
+    font-size: $font-size-lg;
```
Comment (midori0507) [inline src/common/components/modal/modal.module.scss]: same here

File: src/common/components/modal/modal.module.scss [hunk 4]
```diff
@@ -5,6 +5,7 @@
 @import 'variables';
 
 $modalPadding: 2rem;
+$lgModalPadding: 3rem;
```
Comment (midori0507) [inline src/common/components/modal/modal.module.scss]: this is quite unnecessary, just use $largePadding

File: src/common/components/modal/modal.module.scss [hunk 5]
```diff
@@ -32,35 +36,42 @@ $modalPadding: 2rem;
   justify-content: center;
 }
 
+.closeButton {
+  float: right;
+  @include respond-below(md) {
+    z-index: 1;
+    position: relative;
+    text-align: right;
```
Comment (midori0507) [inline src/common/components/modal/modal.module.scss]: this is not needed no more

File: src/domain/child/modal/childFormModal.module.scss [hunk 6]
```diff
@@ -33,13 +33,19 @@
   }
   .submitButton {
     @include button();
+    @include respond-below(sm) {
+      grid-row: 1/2;
```
Comment (midori0507) [inline src/domain/child/modal/childFormModal.module.scss]: `/2` is not needed

File: src/domain/child/modal/childFormModal.module.scss [hunk 7]
```diff
@@ -33,13 +33,19 @@
...
+    }
   }
 
   .cancelButton {
     @include button();
     button {
       background: $hel-light;
     }
+    @include respond-below(sm) {
+      grid-row: 2/2;
```
Comment (midori0507) [inline src/domain/child/modal/childFormModal.module.scss]: `/2` is not needed

File: src/domain/profile/children/ProfileChildrenList.tsx [hunk 8]
```diff
@@ -87,6 +75,18 @@ const ProfileChildrenList: React.FunctionComponent = () => {
           </div>
         )}
       </div>
+      <button
```
Comment (midori0507) [inline src/domain/profile/children/ProfileChildrenList.tsx]: can we use Button component here

File: src/domain/profile/children/profileChildrenList.module.scss [hunk 9]
```diff
@@ -10,26 +10,21 @@
...
-    display: inline-flex;
-    span {
-      margin-right: $baseMargin;
-    }
-    img {
-      transform: rotate(45deg);
-    }
+  display: inline-flex;
+  span {
+    margin-left: 0.2rem;
```
Comment (midori0507) [inline src/domain/profile/children/profileChildrenList.module.scss]: maybe $baseMargin, make var and put top of file

File: src/domain/profile/children/profileChildrenList.module.scss [hunk 10]
```diff
@@ -10,26 +10,21 @@
...
+    font-weight: bold;
```
Comment (midori0507) [inline src/domain/profile/children/profileChildrenList.module.scss]: use $font-weight-bold

File: src/domain/profile/profile.module.scss [hunk 11]
```diff
@@ -26,22 +26,38 @@
...
+    @include respond-below(md) {
+      align-self: start;
+      span {
         display: none;
       }
     }
+    @include respond-above(md) {
+      align-self: center;
+      span {
+        margin-right: 1em;
```
Comment (midori0507) [inline src/domain/profile/profile.module.scss]: $baseMargin

File: src/domain/profile/profile.module.scss [hunk 12]
```diff
@@ -26,22 +26,38 @@
...
   justify-content: space-between;
+  margin-top: $baseMargin;
+  h1 {
+    margin: 0;
+  }
 
   .editProfile {
-    justify-self: right;
-    align-self: center;
+    height: var(--hds-text-xxxl); // FIXME: See if this is actually needed.
```
Comment (midori0507) [inline src/domain/profile/profile.module.scss]: nope

File: src/common/components/modal/modal.module.scss (hunk 4 above)
Comment (badrange) [reply src/common/components/modal/modal.module.scss]: Ah, yes!

File: src/common/components/modal/modal.module.scss (hunk 1 above)
Comment (badrange) [reply src/common/components/modal/modal.module.scss]: I set it to 800px, which should be a suitable value. We only have one modal size at the moment (for md and smaller the modals are responsive/fullwidth).

File: src/common/components/modal/modal.module.scss (hunk 5 above)
Comment (badrange) [reply src/common/components/modal/modal.module.scss]: With the current implementation (float right) you need text-align to make it stay to the right. 
I'll change it to flex.

File: src/common/components/modal/modal.module.scss [hunk 13]
```diff
@@ -32,35 +36,39 @@ $modalPadding: 2rem;
   justify-content: center;
 }
 
+.closeButton {
+  display: flex;
+  justify-content: flex-end;
```
Comment (midori0507) [inline src/common/components/modal/modal.module.scss]: lol nice, thank you!

File: src/domain/profile/modal/editProfileModal.module.scss [hunk 14]
```diff
@@ -22,14 +24,20 @@
...
   button {
     width: 100%;
-    background-color: $hel-gray;
+    background-color: $hel-light;
   }
 }
 
 .submitButton {
+  @include respond-below(sm) {
+    grid-row: 1/2;
```
Comment (midori0507) [inline src/domain/profile/modal/editProfileModal.module.scss]: You missed this one?

File: src/domain/profile/modal/editProfileModal.module.scss [hunk 15]
```diff
@@ -22,14 +24,20 @@
 }
 
 .cancelButton {
+  @include respond-below(sm) {
+    grid-row: 2/2;
```
Comment (midori0507) [inline src/domain/profile/modal/editProfileModal.module.scss]: And this one

File: src/common/components/modal/modal.module.scss (hunk 5 above)
Comment (midori0507) [reply src/common/components/modal/modal.module.scss]: fair enough

File: src/common/components/modal/modal.module.scss (hunk 1 above)
Comment (midori0507) [reply src/common/components/modal/modal.module.scss]: 👍

**Pull request merged**